cloudify_agent/api/pm/base.py

```diff
@@ -485,6 +485,7 @@ def stop(self,
         shutdown.
 
         """
+        self.before_self_stop()
```

Comment:
we're adding `before_self_stop` to `stop` itself, because it must _always_ be called together. It used to be that real usage did remember to call it, but tests forgot. So, we're making just `stop` call it so that the caller doesn't need to remember.